docs/self-managed/setup/deploy/azure/azure-aks/terraform-setup.md

```diff
@@ -0,0 +1,465 @@
+
+The module will be locally sourced, meaning within your cloned repository you can make adjustments to the AKS module directly, and the changes will immediately apply to your setup.
+
+#### Set up the AKS cluster module
+
+1. The `cluster.tf` in your chosen reference contains a basic setup referencing a local Terraform module with the cluster configuration. The following shows said file, which you can adjust within your cloned setup to suit your needs.
+
+```hcl reference
+https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks/cluster.tf
+```
+
+2. Configure user access to the cluster. By default, the identity that creates the Azure AKS cluster has administrative access via role-based access control (RBAC).
+
+<details>
+  <summary>Grant cluster access to other users</summary>
+  <p>
+
+To grant other users access to the cluster, you can assign them Azure roles using Azure Active Directory (Entra ID) integration and Kubernetes RBAC.
+
+This architecture does not currently enable Entra ID integration, but you can manage Kubernetes access through the `az role assignment` and `kubectl` tools:
+
+1. Ensure the user has access to the Azure resource group or subscription, typically via the `Azure Kubernetes Service RBAC Cluster Admin` or `Contributor` role.
+
+2. Then, apply Kubernetes RoleBindings or ClusterRoleBindings to control in-cluster permissions:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: reader-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: view
+subjects:
+  - kind: User
+    name: "<user-object-id>"
+    apiGroup: rbac.authorization.k8s.io
+```
+
+Replace `<user-object-id>` with the Azure AD object ID of the user.
+
+For more detailed instructions, refer to the [AKS access control documentation](https://learn.microsoft.com/azure/aks/manage-azure-rbac).
+
+  </p>
+</details>
+
+3. Customize the cluster setup. The module offers various input options that allow you to further customize the cluster configuration. For a comprehensive list of available options and detailed usage instructions, refer to the [AKS module documentation](https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks-cluster/README.md).
+
+### PostgreSQL module setup
+
+:::info Optional module
+
+If you don't want to use this module, you can skip this section. However, you may need to adjust the remaining instructions to remove references to this module.
+
+If you choose not to use this module, you must either provide a managed PostgreSQL service or use the internal deployment by the Camunda Helm chart in Kubernetes.
+
+Additionally, you must delete the `db.tf` file within your chosen reference as it will otherwise create the resources.
```

Comment:
the suggestion is good, the reality is sadly not the case.
Your PR on the deployment-ref side does not contain this change.

Reply:
the idea in EKS is that you have the reference that uses the modules.
Yes you can eventually adjust the modules if you need something specific but not adjust it for the base setup to configure it.
You wouldn't hardcode the value in a module.
Also here if you delete the db.tf it will still be referenced and create resources + likely not work since the resources in `db.tf` are linked somewhere else in the module.